code/game/objects/structures/hybrisa_props.dm

```diff
@@ -0,0 +1,2778 @@
+		return XENO_NO_DELAY_ACTION
+	current_xenomorph.animation_attack_on(src)
+	playsound(src, 'sound/effects/metalhit.ogg', 25, 1)
+	current_xenomorph.visible_message(SPAN_DANGER("[current_xenomorph] slashes at [src]!"),
+	SPAN_DANGER("You slash at [src]!"), null, 5, CHAT_TYPE_XENO_COMBAT)
+	update_health(rand(current_xenomorph.melee_damage_lower, current_xenomorph.melee_damage_upper))
+	return XENO_ATTACK_ACTION
+
+/obj/structure/prop/hybrisa/misc/buildinggreebliessmall/smallvent2
+	name = "wall vent"
+	icon_state = "smallwallvent2"
+
+/obj/structure/prop/hybrisa/misc/buildinggreebliessmall/smallvent3
+	name = "wall vent"
+	icon_state = "smallwallvent3"
+
+/obj/structure/prop/hybrisa/misc/buildinggreebliessmall/computer
+	name = "computer"
+	icon = 'icons/obj/structures/props/hybrisarandomprops.dmi'
+	icon_state = "zcomputermachine"
+	density = TRUE
+
+/obj/structure/prop/hybrisa/misc/metergreen
+	name = "meter"
+	icon = 'icons/obj/structures/props/hybrisarandomprops.dmi'
+	icon_state = "biggreenmeter1"
+
+/obj/structure/prop/hybrisa/misc/elevator_button
+	name = "broken elevator button"
+	icon = 'icons/obj/structures/props/hybrisarandomprops.dmi'
+	icon_state = "broken_elevator_button"
+
+// MISC
+
+/obj/structure/prop/hybrisa/misc/stoneplanterseats
+	name = "concrete seated planter"
+	desc = "A decorative concrete planter with seating attached, the seats are fitted with synthetic leather, they've faded in time.."
+	icon = 'icons/obj/structures/props/64x64_hybrisarandomprops.dmi'
+	icon_state = "planterseats"
+	bound_width = 32
+	bound_height = 64
+	density = TRUE
+	health = 2000
+	anchored = TRUE
+	layer = ABOVE_MOB_LAYER
+	projectile_coverage = 20
+	throwpass = TRUE
+
+/obj/structure/prop/hybrisa/misc/stoneplanterseats/empty
+	name = "concrete planter"
+	desc = "A decorative concrete planter."
+	icon_state = "planterempty"
+
+/obj/structure/prop/hybrisa/misc/concretestatue
+	name = "concrete statue"
+	desc = "A decorative statue with the Weyland-Yutani 'Wings' adorned on it, A corporate brutalist piece of art."
+	icon = 'icons/obj/structures/props/64x64_hybrisarandomprops.dmi'
+	icon_state = "concretesculpture"
+	bound_width = 64
+	bound_height = 64
+	density = TRUE
+	anchored = TRUE
+	unslashable = TRUE
+	unacidable = TRUE
+	explo_proof = TRUE
+	layer = ABOVE_MOB_LAYER
+
+/obj/structure/prop/hybrisa/misc/detonator
+	name = "detonator"
+	desc = "A detonator for explosives, armed and ready."
+	icon_state = "detonator"
+	density = FALSE
+	anchored = TRUE
+	unslashable = TRUE
+	unacidable = TRUE
+	explo_proof = TRUE
+	projectile_coverage = 20
+	throwpass = TRUE
+	var/id = 1
+	var/range = 15
+
+/obj/structure/prop/hybrisa/misc/detonator/attack_hand(mob/user)
+	for(var/obj/item/explosive/plastic/hybrisa/mining/explosive in range(range))
+		if(explosive.id == id)
+			var/turf/target_turf
+			target_turf = get_turf(explosive.loc)
+			var/datum/cause_data/temp_cause = create_cause_data(src, user)
+			explosive.handle_explosion(target_turf,temp_cause)
```

Comment:
This call is missing `dir`, passing `temp_cause` instead of it and causing runtime down the call stack because of missing `cause_data`.

Reply:
```suggestion
			explosive.handle_explosion(target_turf, explosive.dir, temp_cause)
```